Unable to configure Infinispan idempotent consumers

Attempting to use a Infinispan idempotent consumer results in:

```
Caused by: java.lang.ClassNotFoundException: org.infinispan.commons.api.BasicCacheContainer from [Module "deployment.camel-infinispan-test.jar:main" from Service Module Loader]
        at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:205)
        at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:455)
        at org.jboss.modules.ConcurrentClassLoader.performLoadClassChecked(ConcurrentClassLoader.java:404)
        at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:385)
        at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:130)
        ... 15 more

```

--- a/feature-pack/src/main/resources/modules/system/layers/fuse/org/apache/camel/component/infinispan/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/fuse/org/apache/camel/component/infinispan/main/module.xml
@@ -13,7 +13,13 @@
         <exclude path="OSGI-INF/blueprint" />
       </exports>
     </module>
-    <module name="org.infinispan.commons" />
+    <module name="org.infinispan.commons" export="true">
+      <exports>
+        <include path="org/infinispan/commons" />
+        <include path="org/infinispan/commons/api**" />
+        <exclude path="org/infinispan**" />
+      </exports>
+    </module>
     <module name="org.infinispan.client.hotrod" />
     <module name="org.slf4j" />
     <module name="javax.xml.bind.api" />

--- a/itests/standalone/basic/src/main/java/org/wildfly/camel/test/infinispan/InfinispanIntegrationTest.java
+++ b/itests/standalone/basic/src/main/java/org/wildfly/camel/test/infinispan/InfinispanIntegrationTest.java
@@ -29,6 +29,7 @@ import javax.ejb.Stateless;
 import org.apache.camel.ProducerTemplate;
 import org.apache.camel.builder.RouteBuilder;
 import org.apache.camel.component.infinispan.InfinispanConstants;
+import org.apache.camel.component.infinispan.processor.idempotent.InfinispanIdempotentRepository;
 import org.apache.camel.component.mock.MockEndpoint;
 import org.apache.camel.impl.DefaultCamelContext;
 import org.infinispan.manager.CacheContainer;
@@ -285,6 +286,37 @@ public class InfinispanIntegrationTest {
             mockEndpoint.message(1).outHeader(InfinispanConstants.KEY).isEqualTo(CACHE_KEY_NAME);
 
             cacheContainer.getCache().replace(CACHE_KEY_NAME, CACHE_VALUE_BOB);
+
+            mockEndpoint.assertIsSatisfied();
+        } finally {
+            camelctx.stop();
+        }
+    }
+
+    @Test
+    public void testIdempotentConsumer() throws Exception {
+        final InfinispanIdempotentRepository messageIdRepository = new InfinispanIdempotentRepository(cacheContainer, "myProcessorName");
+
+        DefaultCamelContext camelctx = new DefaultCamelContext();
+        camelctx.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                from("direct:start")
+                .idempotentConsumer(simple("${header.messageId}"), messageIdRepository)
+                .to("mock:result");
+            }
+        });
+
+        camelctx.start();
+        try {
+            MockEndpoint mockEndpoint = camelctx.getEndpoint("mock:result", MockEndpoint.class);
+            mockEndpoint.expectedMessageCount(1);
+
+            // Send 5 messages with the same messageId header. Only 1 should be forwarded to the mock:result endpoint
+            ProducerTemplate template = camelctx.createProducerTemplate();
+            for (int i = 0; i < 5; i++) {
+                template.requestBodyAndHeader("direct:start", null, "messageId", "12345");
+            }
 
             mockEndpoint.assertIsSatisfied();
         } finally {

--- a/itests/standalone/extra/src/main/resources/classloading/exported-path-patterns.txt
+++ b/itests/standalone/extra/src/main/resources/classloading/exported-path-patterns.txt
@@ -26,6 +26,7 @@
 	^org/elasticsearch(.*)
 	^org/quartz(.*)
 	^org/infinispan
+	^org/infinispan/commons(.*)
 	^org/infinispan/manager(.*)
 	^org/springframework(.*)
 	^org/xmlsoap(.*)

--- a/modules/etc/baseline/exported-paths.txt
+++ b/modules/etc/baseline/exported-paths.txt
@@ -1195,6 +1195,8 @@ org/elasticsearch/transport/support
 org/elasticsearch/tribe
 org/elasticsearch/watcher
 org/infinispan
+org/infinispan/commons
+org/infinispan/commons/api
 org/infinispan/manager
 org/infinispan/manager/impl
 org/quartz

--- a/modules/etc/generated/wildfly/modules/system/layers/fuse/org/apache/camel/component/infinispan/main/module.xml
+++ b/modules/etc/generated/wildfly/modules/system/layers/fuse/org/apache/camel/component/infinispan/main/module.xml
@@ -13,7 +13,13 @@
         <exclude path="OSGI-INF/blueprint" />
       </exports>
     </module>
-    <module name="org.infinispan.commons" />
+    <module name="org.infinispan.commons" export="true">
+      <exports>
+        <include path="org/infinispan/commons" />
+        <include path="org/infinispan/commons/api**" />
+        <exclude path="org/infinispan**" />
+      </exports>
+    </module>
     <module name="org.infinispan.client.hotrod" />
     <module name="org.slf4j" />
     <module name="javax.xml.bind.api" />

--- a/modules/etc/smartics/camel-modules.xml
+++ b/modules/etc/smartics/camel-modules.xml
@@ -370,7 +370,13 @@
                     <exclude path="OSGI-INF/blueprint"/>
                 </exports>
             </module>
-            <module name="org.infinispan.commons" />
+            <module name="org.infinispan.commons" export="true">
+                <exports>
+                    <include path="org/infinispan/commons"/>
+                    <include path="org/infinispan/commons/api**" />
+                    <exclude path="org/infinispan**"/>
+                </exports>
+            </module>
             <module name="org.infinispan.client.hotrod" />
             <module name="org.slf4j"/>
         </dependencies>

